scipy/spatial/src/distance_wrap.c

```diff
@@ -707,14 +707,14 @@ static PyObject *to_squareform_from_vector_wrap(PyObject *self, PyObject *args)
     return 0;
   }
   NPY_BEGIN_ALLOW_THREADS;
-  n = M_->dimensions[0];
-  elsize = M_->descr->elsize;
+  n = PyArray_DIMS(M_)[0];
+  elsize = PyArray_DESCR(M_)->elsize;
```

Comment:
Not a 100% sure without checking that the `->elsize` isn't still deprecated API.

To verify, you can modify `spatial/setup.py` like so (there are examples in other `setup.py` files, like for `optimize`):
```python
from scipy._build_utils import numpy_nodepr_api

config.add_extension(...,
     **numpy_nodepr_api)
```
This is generally useful to do.

Reply:
(2 more `->elsize` further down)

Reply:
Checking the docs says it's not deprecated API, see `elsize` in https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_DescrNewFromType. So guess we're good here.